### PR TITLE
file format (fixes #399)

### LINKED ISSUE #399: file format
- Handel file type validation 
- display an error message when the wrong type of file is uploaded
- fix server crash issue when wrong type of file is uploaded

### PR DESCRIPTION
**Pull request by thepuskar**
Fixes #issue_number

### Description
Description of changes made.

### CODE REVIEW

File: src/utils/apiFunc.jsx [hunk 1]
```diff
@@ -15,3 +16,8 @@ export const getApi = async (dispatch, url, config = configFunc()) => {
 export const postApi = async (dispatch, url, data, config = configFunc()) => {
   return await axios.post(url, data, config)
 }
+
+export const postCourseAPI = async (url, data, header) => {
```
Comment (lmmrssa) [inline src/utils/apiFunc.jsx:20]: this is just creating new function where postApi is already doing same

File: src/utils/apiFunc.jsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import axios from 'axios'
+import { Axios } from './urlConstants'
 
-export const configFunc = () => {
+export const configFunc = (header) => {
   const userdata = window.localStorage.getItem('userInfo')
   const token = JSON.parse(userdata).token
-  const headers = { 'Content-Type': 'application/json' }
+  const headers = { 'Content-Type': header || 'application/json' }
```
Comment (lmmrssa) [inline src/utils/apiFunc.jsx:7]: in future other header parameters might also be changed so this has been made too specific. it should merge default headers and header passed

File: src/actions/lessonActions.js [hunk 3]
```diff
@@ -36,7 +36,8 @@ export const createLesson =
         const config = {
           headers: {
             'Content-Type': 'multipart/form-data'
-          }
+          },
```
Comment (lmmrssa) [inline src/actions/lessonActions.js:39]: all these are still not using common postApi

File: api/src/middleware/errorHandler.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
+const multer = require('multer')
```
Comment (lmmrssa) [inline api/src/middleware/errorHandler.js:1]: dont see multer used in file

File: api/src/controllers/photoController.js [hunk 5]
```diff
@@ -42,6 +42,7 @@ const getPhotoById = async (req, res) => {
 }
 
 const addphoto = async (req, res) => {
+  console.log(res.status)
```
Comment (lmmrssa) [inline api/src/controllers/photoController.js:45]: this code can be removed

File: api/src/server.js [hunk 6]
```diff
@@ -35,6 +35,7 @@ const PORT = process.env.port || 5000
 
 // middleware
 app.use(express.json())
+
```
Comment (lmmrssa) [inline api/src/server.js:38]: no need of this extra empty line

**Review by lmmrssa — CHANGES_REQUESTED** [review]

**Review by lmmrssa — APPROVED** [review]

**Review by dogi — APPROVED** [review]

**Pull request merged**